types.cc

```diff
@@ -3042,6 +3042,48 @@ static bytes serialize_for_cql_aux(const list_type_impl&, collection_mutation_vi
     return collection_type_impl::pack(tmp.begin(), tmp.end(), tmp.size(), sf);
 }
 
+static bytes serialize_for_cql_aux(const user_type_impl& type, collection_mutation_view_description mut, cql_serialization_format) {
+    assert(type.is_multi_cell());
+    assert(mut.cells.size() <= type.size());
+
+    std::vector<bytes> linearized;
+    std::vector<bytes_view_opt> tmp;
+    tmp.reserve(mut.cells.size());
+
+    size_t curr_field_pos = 0;
+    for (auto&& e : mut.cells) {
+        auto field_pos = deserialize_field_index(e.first);
+        assert(field_pos < type.size());
+
+        // Some fields don't have corresponding cells -- these fields are null.
+        while (curr_field_pos < field_pos) {
+            tmp.push_back(std::nullopt);
+            ++curr_field_pos;
+        }
+
+        if (e.second.is_live(mut.tomb, false)) {
+            auto value_view = e.second.value();
+            if (value_view.is_fragmented()) {
+                const auto& v = linearized.emplace_back(value_view.linearize());
+                tmp.emplace_back(v);
+            } else {
+                tmp.emplace_back(value_view.first_fragment());
+            }
+        } else {
+            tmp.push_back(std::nullopt);
+        }
+
+        ++curr_field_pos;
+    }
+
+    // Trailing null fields
+    while (curr_field_pos++ < type.size()) {
+        tmp.push_back(std::nullopt);
+    }
+
```

Comment:
I see you are resizing it anyway, so operator[] would be a bit simpler.

Reply:
Done